File: Cadena de Hoteles/Hotel/src/main/java/org/example/repositories/LogRepository.java
package org.example.repositories;

import org.example.data.DatabaseManager;

/**
 * Repository para el registro de eventos de sesion en la tabla LogSesion.
 * Centraliza todos los INSERT de logs del sistema para mantener
 * un historico de autenticacion y actividad de usuarios.
 */
public class LogRepository {

    // IDs de TipoEventoSesion en la base de datos
    public static final int TIPO_LOGIN_EXITOSO       = 1;
    public static final int TIPO_LOGIN_FALLIDO       = 2;
    public static final int TIPO_LOGIN_ERROR_INTERNO = 3;

    public static final int TIPO_LOGOUT_EXITOSO       = 4;
    public static final int TIPO_LOGOUT_ERROR_INTERNO = 5;

    public static final int TIPO_REGISTRO_EXITOSO          = 6;
    public static final int TIPO_REGISTRO_FALLIDO          = 7;
    public static final int TIPO_REGISTRO_ERROR_INTERNO    = 8;
    public static final int TIPO_CAMBIO_TELEFONO_EXITOSO   = 9;
    public static final int TIPO_CAMBIO_TELEFONO_ERROR     = 10;
    public static final int TIPO_CAMBIO_CONTRASENA_EXITOSO = 11;
    public static final int TIPO_CAMBIO_CONTRASENA_FALLIDO = 12;
    public static final int TIPO_CAMBIO_CONTRASENA_ERROR   = 13;

    /**
     * Registra un evento de sesion en la tabla LogSesion.
     * Si el INSERT falla, imprime el error en consola pero NO lanza excepcion
     * para que el flujo principal del sistema no se vea interrumpido por un log.
     *
     * @param tipoEventoId   ID del tipo de evento (usar constantes TIPO_* de esta clase).
     * @param usuarioId      ID del usuario autenticado, null si el login fallo o es anonimo.
     * @param loginIntentado username o correo que se intento usar en el login, puede ser null.
     * @param exitoso        true si el evento fue exitoso, false en caso contrario.
     * @param ipOrigen       IP del cliente que origino la peticion, puede ser null.
     * @param userAgent      User-Agent del cliente, puede ser null.
     * @param mensaje        mensaje adicional descriptivo del evento, puede ser null.
     */
    public void registrar(int tipoEventoId, Integer usuarioId, String loginIntentado,
                          boolean exitoso, String ipOrigen, String userAgent, String mensaje) {
        try {
            DatabaseManager.executeUpdate(
                    "INSERT INTO LogSesion " +
                            "(TipoEventoID, UsuarioID, LoginIntentado, Exitoso, IPOrigen, UserAgent, Mensaje, Fecha) " +
                            "VALUES (?, ?, ?, ?, ?, ?, ?, CURRENT_TIMESTAMP)",
                    tipoEventoId,
                    usuarioId,
                    loginIntentado,
                    exitoso ? 1 : 0,
                    ipOrigen,
                    userAgent,
                    mensaje
            );
        } catch (Exception e) {
            System.err.println("[LogRepository] Error al registrar log (tipo=" + tipoEventoId + "): " + e.getMessage());
        }
    }
}

File: Aero Linea/Aerolinea.API/Aerolinea.API/Repositories/LogRepository.cs
using Aerolinea.API.Data;
using Microsoft.Data.SqlClient;

namespace Aerolinea.API.Repositories
{
    /// <summary>
    /// Repositorio para el registro de eventos de sesion en la tabla LogSesion.
    /// Centraliza todos los INSERT de logs del sistema. Si el INSERT falla,
    /// imprime el error en consola pero no lanza excepcion para no interrumpir
    /// el flujo principal.
    /// </summary>
    public class LogRepository
    {
        public const int TipoLoginExitoso = 1;
        public const int TipoLoginFallido = 2;
        public const int TipoLoginErrorInterno = 3;
        public const int TipoLogoutExitoso = 4;
        public const int TipoLogoutErrorInterno = 5;

        public const int TipoRegistroExitoso = 6;
        public const int TipoRegistroFallido = 7;
        public const int TipoRegistroErrorInterno = 8;

        private readonly DbConnectionFactory _connectionFactory;

        public LogRepository(DbConnectionFactory connectionFactory)
        {
            _connectionFactory = connectionFactory;
        }

        /// <summary>
        /// Registra un evento de sesion en la tabla LogSesion.
        /// </summary>
        /// <param name="tipoEventoId">ID del tipo de evento (usar constantes Tipo* de esta clase).</param>
        /// <param name="usuarioId">ID del usuario autenticado, null si el login fallo.</param>
        /// <param name="loginIntentado">Username o correo intentado, puede ser null.</param>
        /// <param name="exitoso">true si el evento fue exitoso.</param>
        /// <param name="ipOrigen">IP del cliente, puede ser null.</param>
        /// <param name="userAgent">User-Agent del cliente, puede ser null.</param>
        /// <param name="mensaje">Mensaje adicional descriptivo, puede ser null.</param>
        public async Task Registrar(int tipoEventoId, int? usuarioId, string? loginIntentado,
                                    bool exitoso, string? ipOrigen, string? userAgent, string? mensaje)
        {
            try
            {
                using var connection = _connectionFactory.CreateConnection();
                await connection.OpenAsync();

                var sql = @"
                    INSERT INTO LogSesion
                        (TipoEventoID, UsuarioID, LoginIntentado, Exitoso, IPOrigen, UserAgent, Mensaje, Fecha)
                    VALUES
                        (@TipoEventoID, @UsuarioID, @LoginIntentado, @Exitoso, @IPOrigen, @UserAgent, @Mensaje, GETDATE())";

                using var command = new SqlCommand(sql, connection);
                command.Parameters.AddWithValue("@TipoEventoID", tipoEventoId);
                command.Parameters.AddWithValue("@UsuarioID", (object?)usuarioId ?? DBNull.Value);
                command.Parameters.AddWithValue("@LoginIntentado", (object?)loginIntentado ?? DBNull.Value);
                command.Parameters.AddWithValue("@Exitoso", exitoso);
                command.Parameters.AddWithValue("@IPOrigen", (object?)ipOrigen ?? DBNull.Value);
                command.Parameters.AddWithValue("@UserAgent", (object?)userAgent ?? DBNull.Value);
                command.Parameters.AddWithValue("@Mensaje", (object?)mensaje ?? DBNull.Value);

                await command.ExecuteNonQueryAsync();
            }
            catch (Exception e)
            {
                Console.Error.WriteLine($"[LogRepository] Error al registrar log (tipo={tipoEventoId}): {e.Message}");
            }
        }
    }
}